app/controllers/DeclarationController.scala

```diff
@@ -54,84 +64,89 @@ class DeclarationController @Inject()(
       }
+
+      erConnector.submitReport(request.pstr, data).flatMap { _ =>
+        emailFuture.map { _ =>
+          Redirect(controllers.routes.ReturnSubmittedController.onPageLoad(waypoints).url)
+        }
+      }
+  }
+
+  private def sendEmail(psaName: String, email: String, taxYear: String, schemeName: String)(
+    implicit request: DataRequest[_], hc: HeaderCarrier): Future[EmailStatus] = {
+    val requestId = hc.requestId.map(_.value).getOrElse(request.headers.get("X-Session-ID").getOrElse(""))
+
+    val submittedDate = formatSubmittedDate(ZonedDateTime.now(ZoneId.of("Europe/London")))
+    val schemeAdministratorType = AdministratorOrPractitioner.Administrator
+
+    val templateParams = Map(
+      "psaName" -> psaName,
+      "schemeName" -> schemeName,
+      "taxYear" -> taxYear,
+      "dateSubmitted" -> submittedDate
+    )
+
+    emailConnector.sendEmail(schemeAdministratorType,
+      requestId,
+      request.loggedInUser.idName,
+      email,
+      config.fileReturnTemplateId,
+      templateParams).map { emailStatus =>
+      auditService.sendEvent(
+        EventReportingSubmissionEmailAuditEvent(
+          request.loggedInUser.idName,
+          schemeAdministratorType,
+          email
+        )
+      )
+      emailStatus
+    }
   }
 
-  //  private def sendEmail(psaName: String, email: String, taxYear: String, schemeName: String)(
-  //    implicit request: DataRequest[_], hc: HeaderCarrier): Future[EmailStatus] = {
-  //    val requestId = hc.requestId.map(_.value).getOrElse(request.headers.get("X-Session-ID").getOrElse(""))
-  //
-  //    val submittedDate = formatSubmittedDate(ZonedDateTime.now(ZoneId.of("Europe/London")))
-  //    val schemeAdministratorType = AdministratorOrPractitioner.Administrator
-  //
-  //    val templateParams = Map(
-  //      "psaName" -> psaName,
-  //      "schemeName" -> schemeName,
-  //      "taxYear" -> taxYear,
-  //      "dateSubmitted" -> submittedDate
-  //    )
-  //
-  //    emailConnector.sendEmail(schemeAdministratorType,
-  //      requestId,
-  //      request.loggedInUser.idName,
-  //      email,
-  //      config.fileReturnTemplateId,
-  //      templateParams).map { emailStatus =>
-  //        auditService.sendEvent(
-  //          EventReportingSubmissionEmailAuditEvent(
-  //            request.loggedInUser.idName,
-  //            schemeAdministratorType,
-  //            email
-  //          )
-  //        )
-  //        emailStatus
-  //      }
-  //  }
-
-  private def declarationData(pstr: String, taxYear: TaxYear, loggedInUser: LoggedInUser) = {
+  private def declarationData(pstr: String, taxYear: TaxYear, loggedInUser: LoggedInUser): JsObject = {
 
     val psaOrPsp = loggedInUser.administratorOrPractitioner match {
       case AdministratorOrPractitioner.Administrator => "PSA"
       case AdministratorOrPractitioner.Practitioner => "PSP"
       case _ => throw new RuntimeException("Unknown user type")
     }
 
+    val common = Json.obj(
```

Comment:
Are these changes desirable? I think this was the pipeline breaking issue culprit.

Reply:
Please see declarationDetails used in pipeline fix here: https://github.com/hmrc/pension-scheme-event-reporting-frontend/blob/1a6030a9277626a3d6f35ad06afd97fec6eb35dd/app/controllers/DeclarationController.scala#L104

Reply:
I don't know, this was Shehzad's code. @shehzad-ismail8 any thoughts on this?